.github/workflows/bean-instantiations.yml

```
@@ -0,0 +1,293 @@
+name: Check Bean Instantiations on Startup and with deferred eager Initialization
+
+on:
+  push:
+    branches: [ develop ]
+    paths:
+      - src/main/java/**
+  pull_request:
+    paths:
+      -  src/main/java/**
+
+jobs:
+  bean-instantiation-check:
+    runs-on: ubuntu-latest
+
+    env:
+      MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH: 9
+      MAX_DEFERRED_CHAIN_LENGTH: 10
+      MIN_INSTANTIATED_BEANS: 20
+      MAX_INSTANTIATED_BEANS: 92
+      MIN_DEFERRED_CHAIN_LENGTH: 1
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v4
+
+      - name: Validate threshold consistency between BeanInstantiationTracer and GitHub Action
+        run: |
+          echo "Validating that thresholds in BeanInstantiationTracer match GitHub Action values..."
+
+          echo "GitHub Action thresholds:"
+          echo "  MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH = $MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH"
+          echo "  MAX_DEFERRED_CHAIN_LENGTH = $MAX_DEFERRED_CHAIN_LENGTH"
+          echo "  MIN_INSTANTIATED_BEANS = $MIN_INSTANTIATED_BEANS"
+          echo "  MAX_INSTANTIATED_BEANS = $MAX_INSTANTIATED_BEANS"
+          echo "  MIN_DEFERRED_CHAIN_LENGTH = $MIN_DEFERRED_CHAIN_LENGTH"
+
+          JAVA_FILE="src/main/java/de/tum/cit/aet/artemis/core/config/BeanInstantiationTracer.java"
+
+          if [ ! -f "$JAVA_FILE" ]; then
+            echo "❌ BeanInstantiationTracer.java not found at expected location: $JAVA_FILE"
+            echo "Searching for the file..."
+            find . -name "BeanInstantiationTracer.java" -type f || echo "File not found anywhere"
+            exit 1
+          fi
+
+          JAVA_STARTUP_MAX=$(grep -E "STARTUP_MAX_DEPENDENCY_CHAIN_THRESHOLD\s*=" "$JAVA_FILE" | sed -E 's/.*=\s*([0-9]+).*/\1/' | head -1)
```

I think the current name is confusing with later comparisons

```suggestion
          JAVA_MAX_STARTUP_DEPENDENCY_CHAIN_LENGTH=$(grep -E "STARTUP_MAX_DEPENDENCY_CHAIN_THRESHOLD\s*=" "$JAVA_FILE" | sed -E 's/.*=\s*([0-9]+).*/\1/' | head -1)
```